### PR TITLE
*: remove v1/alerts/groups API endpoint

### LINKED ISSUE #1508: Remove alert_groups from alertmanager
I remember it being mentioned that we don't want to use this anymore, and as far as I can tell it is exposed via api but the neither the UI or the amtool use it. I also didn't see it mentioned in the proposal of the v2 api.

Is this something we can remove?

removal would be the opposite of http://petstore.swagger.io/?url=https://raw.githubusercontent.com/mxinden/alertmanager/apiv2/api/v2/openapi.yaml

@simonpasquier @mxinden

### PR DESCRIPTION
**Pull request by simonpasquier**
Closes #1508

_I had this change for a while in my backlog, just rebased on master to fix some conflicts._

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -71,6 +71,15 @@ var corsHeaders = map[string]string{
 	"Cache-Control":                 "no-cache, no-store, must-revalidate",
 }
 
+// Alert is the API representation of an alert, which is a regular alert
+// annotated with silencing and inhibition info.
+type Alert struct {
```
Comment (mxinden) [inline api/api.go]: Great! Thanks for moving APIAlert into `api/api.go`.

**Comment by simonpasquier** [comment]
@mxinden done!

**Comment by mxinden** [comment]
In case one of you clicks merge first, would you mind squashing? Makes preparing the CHANGELOG.md easier for the next release.

**Comment by stuartnelson3** [comment]
squash4lyfe

**Comment by simonpasquier** [comment]
@mxinden done

**Comment by mxinden** [comment]
Thanks @simonpasquier!

**Comment by mpursley** [comment]
Hi @simonpasquier @mxinden, sounds like removing the "v1/alerts/groups" endpoint from the V1 API has broken compatibility with some external clients.. e.g. 
"Don't work with the latest version api (v2) alertmanager" - https://github.com/prymitive/karma/issues/115
"Incompatibility with alertmanager 0.16.x" - https://github.com/cloudflare/unsee/issues/273

I wonder if it make sense if we can just put api/v1/alerts/groups back in? (E.g. just re-add the code that was removed from this PR?)

As:
1. This is the only endpoint from the V1 API that was removed, and
2. It sounds like this endpoint was just removed for reasons like "this is a size-able chunk of code that probably hasn't been used for over a year".  Though, there are some remote clients (like unsee and karma) that are using this endpoint, so might be worth putting it back in to support these API clients?

**Comment by roidelapluie** [comment]
Yes +1 for adding it back. We use karma.

**Comment by simonpasquier** [comment]
@mpursley sorry for the pain, if I would have known that that endpoint was used, it wouldn't have been removed probably. The v1 API is deprecated and the plan is to implement something similar with the v2 API to accommodate karma (see https://github.com/prometheus/alertmanager/issues/868).

**Comment by mpursley** [comment]
> @mpursley sorry for the pain, if I would have known that that endpoint was used, it wouldn't have been removed probably. The v1 API is deprecated and the plan is to implement something similar with the v2 API to accommodate karma (see #868).

Yes, understood. It looks like #868 is now updated to "P2", so hopefully that update/fix gets added soon(ish).  Just keep in mind that there will be users out there that are not able to update unsee/karma or alertmanager until this endpoint gets returned or replaced. 

Thx @simonpasquier

**Pull request merged**